go-controller/pkg/libovsdbops/router.go

```diff
@@ -587,6 +587,45 @@ func CreateOrUpdateLogicalRouterStaticRoutesWithPredicate(nbClient libovsdbclien
 	return err
 }
 
+// CreateOrUpdateUniqueLogicalRouterStaticRoutesWithPredicate looks up a logical
+// router static route from the cache based on a given predicate. If it does not
+// exist, it creates the provided logical router static route. If it does, it
+// updates it. The logical router static route is added to the provided logical
+// router.
+// If more than one route matches the predicate on the router, the additional routes are removed.
+func CreateOrUpdateUniqueLogicalRouterStaticRoutesWithPredicate(nbClient libovsdbclient.Client, routerName string,
```

Comment:
sure I was just concerned about changing it for every use case, but I will go and examine all of those to make sure it should be OK

Reply:
looks like in all of the other cases we should be OK. There is one concern about some of the code ignore's policy type (dst-ip or src-ip) so I'll add those where missing. Otherwise we could have a dst-ip route accidentally remove/override a src-ip route.